extensions/amp-viewer-integration/0.1/messaging/messaging.d.ts

```diff
@@ -0,0 +1,48 @@
+export interface AmpViewerMessage {
+  app: string;
+  type: string;
+  requestid: number;
+  name: string;
+  data: any;
+  rsvp?: boolean;
+  error?: string;
+  messagingToken?: string;
+}
+
+export function parseMessage(message: any): AmpViewerMessage | null;
+
+export type RequestHandler = () => void;
```

Comment:
This function has a bunch of args and could return a Promise, right?

Reply:
Good point, updated.